packages/x-components/src/x-modules/facets/service/facets.service.ts

```diff
@@ -103,11 +104,28 @@ export class DefaultFacetsService implements FacetsService {
       this.setFacetGroup({ facetId: facet.id, groupId: facetsGroup.id });
       this.setFacet(facet);
     });
-    const newFilters = facetsGroup.facets.flatMap(facet => facet.filters);
+    const newFilters = this.flatFilters(facetsGroup);
     this.setFilters(newFilters);
     return newFilters;
   }
 
+  /**
+   * This function returns the filters of the facets group flattened in an array. It keeps the
+   * relations between the filters (parent--children).
+   *
+   * @remarks If it is necessary to deal with more cases than the hierarchical, we need to refactor
```

Comment:
It fails to use `@privateremarks` or `@privateRemarks`:
<img width="783" alt="image" src="https://user-images.githubusercontent.com/5759712/181012519-b8110912-edce-4361-a49c-a356198100b6.png">

Reply:
This is a[ valid tag for API extractor](https://api-extractor.com/pages/tsdoc/tag_privateremarks/), which is a TypeScript tool. As we use the jsdoc eslint plugin, it complains as it is not part of the jsdoc standard. Fortunately, the rule that check this is extendable, so we can add it [here](https://github.com/empathyco/x/blob/main/packages/eslint-plugin-x/lib/configs/jsdoc.js#L13).

Reply:
done